include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //VkBool32              integerDotProduct64BitUnsignedAccelerated;
+    //VkBool32              integerDotProduct64BitSignedAccelerated;
+    //VkBool32              integerDotProduct64BitMixedSignednessAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating8BitUnsignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating8BitSignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating8BitMixedSignednessAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating4x8BitPackedUnsignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating4x8BitPackedSignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating4x8BitPackedMixedSignednessAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating16BitUnsignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating16BitSignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating16BitMixedSignednessAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating32BitUnsignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating32BitSignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating32BitMixedSignednessAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating64BitUnsignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating64BitSignedAccelerated;
+    //VkBool32              integerDotProductAccumulatingSaturating64BitMixedSignednessAccelerated;
+    //VkDeviceSize          storageTexelBufferOffsetAlignmentBytes;
+    //VkBool32              storageTexelBufferOffsetSingleTexelAlignment;
+    //VkDeviceSize          uniformTexelBufferOffsetAlignmentBytes;
+    //VkBool32              uniformTexelBufferOffsetSingleTexelAlignment;
+    //VkDeviceSize          maxBufferSize;
+
+    /* Vulkan Extensions */
+
+    /* ShaderCorePropertiesAMD *//* provided by VK_AMD_shader_core_properties */
+    //uint32_t           shaderEngineCount;
+    //uint32_t           shaderArraysPerEngineCount;
+    //uint32_t           computeUnitsPerShaderArray;
+    //uint32_t           simdPerComputeUnit;
+    //uint32_t           wavefrontsPerSimd;
+    //uint32_t           wavefrontSize;
+    //uint32_t           sgprsPerSimd;
+    //uint32_t           minSgprAllocation;
+    //uint32_t           maxSgprAllocation;
+    //uint32_t           sgprAllocationGranularity;
+    //uint32_t           vgprsPerSimd;
+    //uint32_t           minVgprAllocation;
+    //uint32_t           maxVgprAllocation;
+    //uint32_t           vgprAllocationGranularity;
+
+    /* ShaderCoreProperties2AMD *//* provided by VK_AMD_shader_core_properties2 */
+    //VkShaderCorePropertiesFlagsAMD    shaderCoreFeatures;
+    //uint32_t                          activeComputeUnitCount;
```

Comment:
we will never expose this vendor specific crap to the user, but might use the extension to provide some cross platform meta-info in the Nabla section 
https://github.com/Devsh-Graphics-Programming/Nabla/pull/357/files#r888127037